File: GFG_streak/sort-0-1-2.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
class Solution {
  public:
    void sort012(vector<int>& arr) {
        // code here
        int i = 0, f = 0, b = arr.size()-1;
        while(f <= b){
            if(arr[f] == 0){
                if(f==i){
                    i++;
                    f++;
                }
                else{
                    swap(arr[i++], arr[f]);
                }
            }
            else if(arr[f] == 1){
                f++;
            }
            else{
                swap(arr[f],arr[b--]);
            }
        }
    }
};

//{ Driver Code Starts.
int main() {

    int t;
    cin >> t;
    cin.ignore();
    while (t--) {

        vector<int> a;
        string input;
        int num;

        getline(cin, input);
        stringstream s2(input);
        while (s2 >> num) {
            a.push_back(num);
        }
        Solution ob;
        ob.sort012(a);

        int n = a.size();
        for (int i = 0; i < n; i++) {
            cout << a[i] << " ";
        }

        cout << endl;
        cout << "~" << endl;
    }
    return 0;
}

// } Driver Code Ends

File: LC_PYTHON/GFG_streak/sort-0-1-2.py
#{ 
 # Driver Code Starts

# } Driver Code Ends
class Solution:
    # Function to sort an array of 0s, 1s, and 2s
    def sort012(self, arr):
        # code here
        i = 0
        f = 0
        b = len(arr)-1
        while f<=b:
            if arr[f] == 0:
                if i==f:
                    i+=1
                    f+=1
                else:
                    arr[i], arr[f] = arr[f], arr[i]
                    i+=1
            elif arr[f] == 1:
                f+=1
            else:
                arr[b], arr[f] = arr[f], arr[b]
                b-=1

#{ 
 # Driver Code Starts.
def main():
    t = int(input().strip())  # Read the number of test cases
    ob = Solution()

    while t > 0:
        t -= 1
        arr = list(map(int,
                       input().strip().split())
                   )  # Read the array as space-separated integers
        ob.sort012(arr)  # Sort the array

        print(' '.join(map(str, arr)))  # Print the sorted array
        print("~")
        
if __name__ == "__main__":
    main()

# } Driver Code Ends